receiver/skywalkingreceiver/internal/trace/tracing_report_service.go

```diff
@@ -0,0 +1,143 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/skywalkingreceiver"
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/obsreport"
+	"go.opentelemetry.io/collector/receiver"
+	"google.golang.org/protobuf/proto"
+	common "skywalking.apache.org/repo/goapi/collect/common/v3"
+	agent "skywalking.apache.org/repo/goapi/collect/language/agent/v3"
+	v3 "skywalking.apache.org/repo/goapi/collect/language/agent/v3"
+)
+
+const (
+	collectorHTTPTransport = "http"
+	grpcTransport          = "grpc"
+	failing                = "failing"
+)
+
+type Receiver struct {
+	nextConsumer consumer.Traces
+	grpcObsrecv  *obsreport.Receiver
+	httpObsrecv  *obsreport.Receiver
+	agent.UnimplementedTraceSegmentReportServiceServer
+}
+
+// New creates a new Receiver reference.
+func New(nextConsumer consumer.Traces, set receiver.CreateSettings) (*Receiver, error) {
```

Comment:
how about rename `New` as `NewReceiver`?

Reply:
make sense. I'll rename it